### ISSUE TITLE
possible erroneous warning (missing field initializer)

### ISSUE THREAD

**Issue by chardan** [llvm/llvm-project#209270]
I think this isn't a valid warning:
possible-warning-bug.cpp:22:47: warning: missing field 'bar' initializer

The compiler warns that `bar` is missing from the aggregate initializer, even though (as I understand them)
the C++ aggregate-initialization rules default-construct it as an empty std::string.

[possible-warning-bug.cpp](https://github.com/user-attachments/files/29979676/possible-warning-bug.cpp)

**Comment by erichkeane** [comment]
Text of the cpp file:

```c++
/*
Compile with:

 g++ -std=c++26 -Wall -Wextra -pedantic -fsyntax-only possible-warning-bug.cpp
 clang++ -std=c++26 -Wall -Wextra -pedantic -fsyntax-only possible-warning-bug.cpp

Both compilers warn that `bar` is missing from the aggregate initializer, even
though the C++ aggregate-initialization rules default-construct it as an empty
std::string.
*/

#include <string>

struct bletch
{
 int foo;
 std::string bar;
};

int main()
{
 [[maybe_unused]] const bletch wtf { .foo = 0 };
```

Godbolt link:
https://godbolt.org/z/1WMzK7Gsb

**Comment by erichkeane** [comment]
GCC has a thread here where it is claimed this is intentional behavior (initializer vs initialized, see comment 3 in particular).  https://gcc.gnu.org/bugzilla/show_bug.cgi?id=96868

**Comment by chardan** [comment]
This does indeed seem to be that same "non-issue"-- it's just (very) suprising behavior. I will close the bug.
